pyproject.toml

```diff
@@ -32,6 +32,7 @@ android_backup = { version = "^0", optional = true }
 importlib_metadata = { version = "^1", markers = "python_version <= '3.7'" }
 croniter = ">=1"
 defusedxml = "^0"
+dataclasses = "*"
```

Comment:
As we already depend on [attrs](https://www.attrs.org/), we should not use dataclasses which is sort of attrs-"lite" that was added to the stdlib.

Reply:
Personally I prefer dataclasses over attr, it's super lightweight and is a standard lib in >=3.7 making it one less library to include. It's only on here as a backport for 3.6 to make tox pass.
[PEP 557](https://www.python.org/dev/peps/pep-0557/)

Reply:
I have been previously bitten by such compat libs that caused breakages down the path (see that importlib_metadata just a couple of lines above), I'd much prefer not going that route again. 

And considering that debian oldstable was shipping 3.7, I don't see no reason whatsoever constrain to 3.6. That being said, attrs is included as a dep already so there is no reason not to use it and it's pretty much drop-in replacement. And `@attr.s(auto_attribs=True)` can be used for simplest container classes to even make the code look more like dataclasses definitions without the `foo = attr.ib()` definitions.

Reply:
Yeah technically it should have a <3.7 python tag on the import.

I'm all for killing off old python versions ASAP, things just keep getting better and better and it's such a pain being constrained to backports (if they even exist) or old ways of doing thing.
PEP 634 will take such a long time to adopt yet it's something people have been screaming for for ages.

Same as my dislike of pytz and it's inability to reliably return valid tzinfo objects, this is fixed in 3.9 with ZoneInfo but no backport and it'll be a long time until 3.9 is ubiquitous.

But anywho back on track,  if you wish to refactor to use attrs be my guest, but personally I don't want to go read up all the intricacies of a package I don't use and who's functionality is largely being baked into stdlibs by means of dataclasses.